src/jormungandr/explorer/src/api/graphql/mod.rs

```diff
@@ -1354,22 +1355,17 @@ impl VotePlanStatus {
             proposals: proposals
                 .into_iter()
                 .map(|proposal| VoteProposalStatus {
-                    proposal_id: ExternalProposalId::from(proposal.proposal_id),
-                    options: VoteOptionRange::from(proposal.options),
-                    tally: proposal.tally.map(|tally| match tally {
-                        ExplorerVoteTally::Public { results, options } => {
-                            TallyStatus::Public(TallyPublicStatus {
-                                results: results.iter().map(Into::into).collect(),
-                                options: options.into(),
-                            })
-                        }
-                        ExplorerVoteTally::Private { results, options } => {
-                            TallyStatus::Private(TallyPrivateStatus {
-                                results: results.map(|res| res.iter().map(Into::into).collect()),
-                                options: options.into(),
-                            })
-                        }
-                    }),
+                    proposal_id: ExternalProposalId::from(proposal.proposal_id.clone()),
+                    options: VoteOptionRange::from(proposal.options.clone()),
+                    tally: proposal.tally.clone().map_or(
+                        generic_tally_status(ExplorerVoteProposal {
+                            proposal_id: proposal.proposal_id,
+                            options: proposal.options,
+                            tally: proposal.tally,
+                            votes: proposal.votes.clone(),
+                        }),
+                        |tally| to_tally_status(Some(tally)),
```

Comment:
This would turn into `TallyStatus::from`

Reply:
thanks, yes that is much nicer. I have added this, can you double check recent addition to confirm I understood correctly?

Reply:
Yes! Nice :)